tools/pgroonga-primary-maintainer.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env bash
+
+set -xueo pipefail
+
+WAL_BLOCK_SIZE=8192
+
+numfmt_available=0
+if type numfmt >/dev/null 2>&1; then
+  numfmt_available=1
+fi
+
+reindex_threshold_size=0
+psql_command=psql
+psql_database_name=""
+psql_options=""
+
+function usage () {
+  size_option_example="--thresholds 10M, -s 1G"
+  if [ ${numfmt_available} -eq 0 ]; then
+    size_option_example="--thresholds 10485760"
+  fi
+  cat <<USAGE
+$0 --thresholds REINDEX_THRESHOLD_SIZE [--psql PSQL_COMMAND_PATH] [--dbname DATABASE_NAME] [--psql_options PSQL_COMMAND_OPTIONS]
+
+Options:
+-t, --thresholds:
+  If the specified value is exceeded, \`REINDEX INDEX CONCURRENTLY\` is run.
+  Specify by size.
+  Example: ${size_option_example}
+-c, --psql:
+  Specify the path to \`psql\` command.
+-d, --dbname:
+  Specify the database name.
+-o, --psql_options:
+  \`psql\` command options.
+  Example: --psql_options "-h example.com -p 5432"
+-h, --help:
+  Display help text and exit.
+USAGE
+}
+
+function run_psql () {
+  sql="${1}"
+  ${psql_command} \
+    ${psql_options} \
+    --tuples-only \
+    --command "${sql}"
+}
+
+short_options="t:c:d:o:h"
+long_options="thresholds:,psql:,dbname:,psql_options:,help"
+if [ "$(getopt --help)" = " --" ]; then
+ options=$(getopt "${short_options}" "$@")
+else
+ options=$(
+   getopt \
+     --options "${short_options}" \
+     --longoptions "${long_options}" \
+     --name "${0}" \
+     -- "$@"
+ )
+fi
+eval set -- "$options"
+
+while [[ $# -gt 0 ]]; do
+  case "${1}" in
+    -t|--thresholds)
+      if [ ${numfmt_available} -eq 1 ]; then
+        reindex_threshold_size=$(numfmt --from iec "${2}")
+      else
+        reindex_threshold_size="${2}"
+      fi
+      shift 2
+      ;;
+    -c|--psql)
+      psql_command="${2}"
+      shift 2
+      ;;
+    -d|--dbname)
+      psql_database_name="${2}"
+      shift 2
+      ;;
+    -o|--psql_options)
+      psql_options="${2}"
+      shift 2
+      ;;
+    -h|--help)
+      usage
+      exit 0
+      ;;
+    --)
+      shift
+      break
+      ;;
+    *)
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+type "${psql_command}" || (echo 'No psql command.' && exit 1)
```

Comment:
Do we need this?

Reply:
We get an error if we don't have the `psql` command, In that case I thought it would be better to exit with an error early.

Reply:
If we want to keep this check, could you use `if ...; then; echo "error message"; exit 1; fi` style like other check?

BTW, `if ! "${psql_command}" --help > /dev/null;` may be better.